pvlib/temperature.py

```diff
@@ -0,0 +1,253 @@
+    +---------------+----------------+-------+---------+---------------------+
+
+    References
+    ----------
+    [1] King, D. et al, 2004, "Sandia Photovoltaic Array Performance
+    Model", SAND Report 3535, Sandia National Laboratories, Albuquerque,
+    NM.
+
+    '''
+    module_temperature = sapm_module(poa_global, temp_air, wind_speed,
+                                     a, b)
+    return module_temperature + (poa_global / irrad_ref) * (deltaT)
+
+
+def sapm_module(poa_global, temp_air, wind_speed, a, b):
+    r'''
+    Calculate module back surface temperature per the Sandia PV Array
+    Performance Model [1].
+
+    Parameters
+    ----------
+    poa_global : numeric or Series
+        Total incident irradiance [W/m^2].
+
+    temp_air : numeric or Series
+        Ambient dry bulb temperature [C].
+
+    wind_speed : numeric or Series
+        Wind speed at a height of 10 meters [m/s].
+
+    a : float
+        Parameter :math:`a` in :eq:`sapm1`.
+
+    b : float
+        Parameter :math:`b` in :eq:`sapm1`.
+
+    Returns
+    -------
+    numeric or Series, values in degrees C.
+
+    Notes
+    -----
+    The model for module temperature :math:`T_{m}` is given by Eq. 11 in [1].
+
+    .. math::
+       :label: sapm1
+        T_{m} = E \times \exp (a + b \times WS) + T_{a}
+
+    Inputs to the model are plane-of-array irradiance :math:`E` (W/m2) and
+    ambient air temperature :math:`T_{a}` (C). Model outputs are surface
+    temperature at the back of the module :math:`T_{m}` and cell temperature
+    :math:`T_{C}`. Model parameters depend both on the module construction and
+    its mounting. Parameter sets are provided in [1] for representative modules
+    and mounting.
+
+    +---------------+----------------+-------+---------+---------------------+
+    | Module        | Mounting       | a     | b       | :math:`\Delta T [C]`|
+    +===============+================+=======+=========+=====================+
+    | glass/glass   | open rack      | -3.47 | -0.0594 | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/glass   | close roof     | -2.98 | -0.0471 | 1                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | open rack      | -3.56 | -0.075  | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | insulated back | -2.81 | -0.0455 | 0                   |
+    +---------------+----------------+-------+---------+---------------------+
+
+    References
+    ----------
+    [1] King, D. et al, 2004, "Sandia Photovoltaic Array Performance
+    Model", SAND Report 3535, Sandia National Laboratories, Albuquerque,
+    NM.
+
+    '''
+    return poa_global * np.exp(a + b * wind_speed) + temp_air
+
+
+def pvsyst_cell(poa_global, temp_air, wind_speed=1.0, u_c=29.0, u_v=0.0,
```

Comment:
Should we be concerned with inconsistency between what's an arg and what's a kwarg with default in the sapm vs. pvsyst functions?

Reply:
storing parameters in a dict argues in favor of kwargs in all functions so that the args/kwargs can be consistently splatted with * or **

Reply:
I'm less concerned now that I've reminded myself that this works:

```python
In [4]: def f(a, b):
   ...:     print(a, b)
   ...:

In [5]: f(**{'b': 2, 'a': 1})
1 2
```

Reply:
I assigned default values where there is a sensible choice. In the case of pvsyst, the default behavior in the PVsyst software is to assume wind_speed of 1.0  m/s. I don't think it is essential to duplicate that behavior here, but I think it helps.  For the SAPM temperature model I don't think it is helpful to assume default values for `a` and `b` because that implies that a module construction (e.g., glass/glass) is assumed by default.  Outside of assigning default values I am not educated about the nuances of an arg vs kwarg.

Reply:
I think it's ok to leave as is. I'll suggest an example for the docstring later today.

Reply:
Didn't understand this issue.  Should I?